### ISSUE TITLE
Change the interpolation algorithm used for the preview in the Editor and the thumbnails in the File Browser

### ISSUE THREAD

**Issue by teplit** [Beep6581/RawTherapee#6629]
The developed image in the Editor looks too soft when it is downscaled (zoom level below 100%). A similar softness affects the thumbnails in the File Browser.

Here is a comparison between RawTherapee 5.9 and Camera Raw 15 with a zoom level of 33%.

**RawTherapee 5.9**
![RawTherape 5 9 interpolation](https://user-images.githubusercontent.com/5978882/205364403-a572f807-039b-45dd-bc7e-a2f66caabf8b.png)

**CameraRaw 15**
![Camera Raw 15 interpolation](https://user-images.githubusercontent.com/5978882/205364509-e0565428-e5cf-491f-be94-e516750f8fd3.png)

At 100% zoom level the RawTherapee image actually is sharper than the CameraRaw image.

**RawTherapee 5.9**
![RawTherape 5 9 - 100% view](https://user-images.githubusercontent.com/5978882/205367717-5d6e32cd-9ce3-4746-8474-a4dcbe8c694b.jpg)

**CameraRaw 15**
![Camera Raw 15 - 100% view](https://user-images.githubusercontent.com/5978882/205367757-4f752b24-dffc-4802-a76c-46525c787caa.jpg)

The RAW file is taken from [PlayRaw](https://discuss.pixls.us/t/just-a-typical-dutch-scene/19143).

**Comment by Thanatomanic** [comment]
Please take a look at this thread: https://discuss.pixls.us/t/raw-preview-in-editor-window-is-blurry/25514/3. Please let us know your thoughts on what should be done, and why.

**Comment by teplit** [comment]
I read the discussion. I’m neither a software developer nor I want to be such.

I think that RawTherapee needs hardware acceleration in order to move forward in the future as viable RAW processing software. A GPU acceleration in RawTherapee will open the doors for more computationally expensive algorithms and it will improve the overall usability. It’s not a joy to use RawTherapee with 4K monitor…

I can’t recommend a particular interpolation algorithm – it’s neither my job nor hobby to make and compare measurements of computational footprint for different algorithms or different implementations of a particular algorithm. I know that Lanczos gives good results with acceptable amount of artifacts, but this depends on the implementation. For example in Affinity Photo there are two implementations of Lanczos – separable and non-separable. The latter produces sharper image, but with a lot more ringing artifacts. I’m also aware of some academic papers about image downscaling algorithms ([Rapid, Detail-Preserving Image Downscaling](https://dl.acm.org/doi/pdf/10.1145/2980179.2980239) and [Content-Adaptive Image Downscaling](https://johanneskopf.de/publications/downscaling/paper/downscaling.pdf)), but I don’t know how well they work in practice.

**Comment by Lawrence37** [comment]
I'm not convinced the issue is with the interpolation algorithm. Do you have any lens distortion correction turned on? That will introduce softness because the correction is applied on the reduced resolution preview.

**Comment by teplit** [comment]
Yes, there is applied automatic lens distortion correction. Without it the grass in the scene looks better (sharper) at 33% scaling in RawTherapee. However CameraRaw also applies automatic lens distortion correction:

![Camera Raw - automatic lens correction](https://user-images.githubusercontent.com/5978882/205478620-21edff59-fc28-4115-bf7f-d2f9b1484aa7.png)
